src/doc_types.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "redisearch.h"
+#include "redismodule.h"
+#include "json.h"
+
+extern RedisModuleCtx *RSDummyContext;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static inline DocumentType getDocType(RedisModuleKey *key) {
+  if (RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_HASH) {
+    return DocumentType_Hash;
+  } else if (1) {
```

Comment:
We need to use some RedisJson api to verify if its a json

Reply:
We just added isJSON api for this